keps/sig-scheduling/NNNN-pod-to-node-assignment/README.md

```diff
@@ -0,0 +1,808 @@
+
+## Proposal
+
+The idea is to create a scheduler filter plugin such that given a pod and a
+node, a node is filtered out if any of the following is true (assume the API is
+the annotation k8s.io/pod-per-node=true, more on that below):
+
+-   The incoming pod is not a daemonset pod (by checking OwnerReference) and
+    there is a pod with k8s.io/pod-per-node=true already assigned to the node,
+-   If the incoming pod has the k8s.io/pod-per-node=true annotation and the node
+    hosts at least one non daemonset pod. The above logic would also be added to
+    the set of predicates that kubelet tests on pod admission. This will prevent
+    users from working around the scheduler to avoid 1:1 enforcement.
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+A user wants a Pod to be scheduled alone on a node to avoid them from contending
+on resources (e.g. claiming all CPU cores available on a node). In other words,
+resource intensive workloads are needed to be scheduled on separate nodes.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+This section aims to describe how will the feature look in the pod spec.
+Following are some of the options that could be considered.
+
+### Introduce a new API Field
+
+A new API filed `pod.spec.nodeAsssignmentPolicy` could be introduced to be used
+by the new filter. The value for the newly introduced field could be an enum
+that has one of the following values: PodPerNode, Default where default is the
+same as the current behaviour and PodPerNode would enforce 1:1 pod to node
+mapping during scheduling of a given pod.
+
+```yaml
+...
+spec:
+  nodeAssignmentPolicy: <string>
+```
+
+### Define a NodeSelector/NodeAffinity Constraint
```

Comment:
I think this option is obscure with edges cases and so can cause confusion to users. I would move it to "Alternatives"